File: merge-strings-alternately.swift
class Solution {
    func mergeAlternately(_ word1: String, _ word2: String) -> String {
        var result = ""

        for (ch1, ch2) in zip(word1, word2){
            result += [ch1, ch2]
        }

        if word1.count < word2.count {
            result += word2[word1.endIndex...]
        }
        else {
            result += word1[word2.endIndex...]
        }

        return result
    }
}

File: js/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
  n = word1.length > word2.length ? word1.length : word2.length;
  newWord = []

  for(let i = 0; i < n; i++) {
      if (word1[i] !== undefined) {
          newWord += word1[i]
      }
      if (word2[i] !== undefined) {
          newWord += word2[i]
      }
  }

  return newWord;
};